### PR TITLE
Releases v0.14.1 and prepares development

### PR DESCRIPTION
**Pull request by johnedquinn**
## Relevant Issues
- None

## Description
- **IMPORTANT**: Please review both commits. I will tag the first commit with `v0.14.1`.

## Other Information
- Updated Unreleased Section in CHANGELOG: **YES**
- Any backward-incompatible changes? **NO**
- Any new external dependencies? **NO**
- Do your changes comply with the [Contributing Guidelines](https://github.com/partiql/partiql-lang-kotlin/blob/main/CONTRIBUTING.md)
  and [Code Style Guidelines](https://github.com/partiql/partiql-lang-kotlin/blob/main/CODE_STYLE.md)? **YES**

## License Information

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by am357 — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -46,6 +64,7 @@ Thank you to all who have contributed!
 Thank you to all who have contributed!
 - @rchowell
 - @johnedquinn
+- @yliuuuu
```
Comment (yliuuuu) [inline CHANGELOG.md:67]: Thanks : )

**Review by yliuuuu — APPROVED** [review]

File: gradle.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
 group=org.partiql
-version=0.14.1-SNAPSHOT
+version=0.14.2-SNAPSHOT
```
Comment (yliuuuu) [inline gradle.properties:2]: Should this be 0.14.1?

**Review by yliuuuu — CHANGES_REQUESTED** [review]
One question on the version. Sorry I missed this.

File: gradle.properties (hunk 2 above)
Comment (johnedquinn) [reply gradle.properties:2]: Please see the PR description

**Review by yliuuuu — APPROVED** [review]

**Pull request merged**